llmfoundry/utils/config_utils.py

```diff
@@ -178,9 +183,136 @@ def log_config(cfg: DictConfig) -> None:
+    should be categorized as. Possible data sources are Delta Tables, UC Volumes,
+    HuggingFace paths, remote storage, or local storage.
+
+    Args:
+        cfg (DictConfig): A config dictionary of a run
+
+    Returns:
+        List[Tuple[str, str, str]]: A list of tuples formatted as (data type, path, split)
+    """
+    data_paths = []
+
+    # Handle train loader if it exists
+    train_data_set = cfg.get('train_loader', {}).get('dataset', {})
+    train_split = train_data_set.get('split', None)
+    train_source_path = cfg.get('source_dataset_train', None)
+    _process_data_source(train_source_path, train_data_set, train_split,
+                         'train', data_paths)
+
+    # Handle eval_loader which might be a list or a single dictionary
+    eval_data_loaders = cfg.get('eval_loader', {})
+    if not isinstance(eval_data_loaders, ListConfig):
+        eval_data_loaders = [eval_data_loaders
+                            ]  # Normalize to list if it's a single dictionary
+
+    for eval_data_loader in eval_data_loaders:
+        eval_data_set = eval_data_loader.get('dataset', {})
+        eval_split = eval_data_set.get('split', None)
+        eval_source_path = cfg.get('source_dataset_eval', None)
+        _process_data_source(eval_source_path, eval_data_set, eval_split,
+                             'eval', data_paths)
+
+    return data_paths
+
+
+def _process_data_source(source_dataset_path: Optional[str],
+                         data_set: Dict[str, str], cfg_split: Optional[str],
+                         true_split: str, data_paths: List[Tuple[str, str,
+                                                                 str]]):
+    """Add a data source by mutating data_paths.
+
+    Given various dataset attributes, attempt to determine what type of dataset is being added, and parse
+    the dataset accordingly.
+
+    source_dataset_path (Optional[str]): The source dataset in a cfg metadata
+    data_set (Dict[str, str]): The dataset from cfg
+    cfg_split (str): The split listed for the dataset in cfg
+    true_split (str): The split of the dataset to be added
+    data_paths (List[Tuple[str, str, str]]): A list of tuples formatted as (data type, path, split)
+    """
+    # Check for Delta table
+    if source_dataset_path and len(source_dataset_path.split('.')) >= 3:
+        data_paths.append(('delta_table', source_dataset_path, true_split))
+    # Check for UC volume
+    elif source_dataset_path and source_dataset_path.startswith('dbfs:'):
+        data_paths.append(
+            ('uc_volume', source_dataset_path[len('dbfs:'):], true_split))
+    # Check for HF path
+    elif 'hf_name' in data_set:
+        hf_path = data_set['hf_name']
+        backend, _, _ = parse_uri(hf_path)
+        if backend:
+            hf_path = f'{hf_path.rstrip("/")}/{cfg_split}' if cfg_split else hf_path
+            data_paths.append((backend, hf_path, true_split))
+        elif hf_path.startswith('/'):
```

Comment:
local path doesn't have to start with `/`, can we check `os.path.exists` here instead?